src/core/router/atoms/currentLocation.ts

```
@@ -2,11 +2,15 @@ import { createAtom } from '~utils/atoms';
 
 export const currentLocationAtom = createAtom(
   {
-    set: (location: Location) => location,
+    set: (location: CurrentLocation) => location,
   },
-  ({ onAction }, state = globalThis.location) => {
+  ({ onAction }, state: CurrentLocation = globalThis.location) => {
     onAction('set', (location) => (state = location));
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)